app/models/ldap_identity_provider.rb

```diff
@@ -56,8 +58,12 @@ def ldap_search(filter:)
       size: 500,
       return_results: false
     ) do |entry|
-      if new_user = ldap_entry_to_user(entry)
-        results << new_user
+      if valid_ldap_entry? entry
+        if results.length < affiliates_offset.to_i || affiliates_limit && results.length >= affiliates_limit + affiliates_offset.to_i
```

Comment:
could you do something like
```
results = Array.new(500)
iter = 0
success = ldap_conn.search(..) do |entry|
  if valid_ldap_entry? entry
    limit = affiliates_limit ? affiliates_limit + affiliates_offset.to_i : results.length
    if iter >= affiliates_limit && iter <= limit
     results << ldap_entry_to_user(entry)
    end
  end
  iter = iter + 1
end
```

Reply:
I could... But why?

Reply:
simpler? no results << nil.

Reply:
Oh, no. I see what you're doing. That would end up breaking the pagination in the request, which takes the results of `#affiliates` and passes it to `Kaminari.paginate_array`, then to the `paginate` method.

Reply:
I'm not sure I follow, but it doesnt matter since this is iteration one anyway.